backend/src/stwo/circuit_builder.rs

```diff
@@ -122,53 +149,67 @@ fn to_stwo_expression<T: FieldElement, E: EvalAtRow>(
     witness_eval: &Vec<[<E as EvalAtRow>::F; 2]>,
     _eval: &E,
```

Comment:
If this is unused it should be removed

Reply:
Done. Now I know I can simply use "function_name::<T, E>(...)" as explicit type parameter specification, didn't mange to find this solution before, Rust always tell me to give type annotations if I don't pass this unused _eval ^^'

Reply:
Having to use explicit type parameters is often a hint that the generics can be simplified. In this case, the function is generic over `E` but the signature only uses `E::F`. This requires explicit type parameters, because you pass a `F` but there could be many `E` which have this `F` as their `E::F`, and the compiler needs to know which `E` you're talking about. I suspect it could be possible to change function to be generic over `F`, in which case we would not need explicit type parameters anymore.

Reply:
Done. Nice tip to help me discover potential simplification, thanks!